src/lib.rs

```diff
@@ -1,90 +1,116 @@
+pub mod error;
+pub mod support;
+
+#[macro_use]
+mod macros;
+
 extern crate xenctrl_sys;
-use std::io::Error;
-use std::ptr::{null_mut};
-use std::os::raw::{c_void, c_ulong};
+use std::{
+    mem,
+    ptr::{null_mut, NonNull},
+};
+
+use xenctrl_sys::{
+    xc_clear_last_error, xc_domain_maximum_gpfn, xc_domain_pause, xc_domain_unpause, xc_error_code,
+    xc_get_last_error, xc_interface, xc_interface_close, xc_interface_open, xc_monitor_disable,
+    xc_monitor_enable, xentoollog_logger,
+};
 
-pub const PAGE_SHIFT: u32 = xenctrl_sys::XC_PAGE_SHIFT;
```

Comment:
You need to export these constants as well

Reply:
Sorry, I forgot this. PR updated.